### ISSUE TITLE
Match kotlinc behavior for Repository annotation

### ISSUE THREAD

**Issue by gabrielfeo** [kscripting/kscript#388]
In a Kotlin script with external dependencies, kotlinc supports adding multiple repositories with a single annotation ([source][1]):

```kotlin
@file:Repository(
  "https://jitpack.io",
  "https://repo.maven.apache.org/maven2",
)
@file:DependsOn(
  "a-jitpack-artifact",
  "a-maven-central-artifact",
```

However, kscript parses differently, as if the second arg were a repository username:

```
[kscript] Adding repository: Repository(id=, url=https://jitpack.io, user=https://repo.maven.apache.org/maven2, password=)
```

It'd be great if kscript could match behavior, to be 100% compatible with the KEEP-75 annotations. Even if one writes a script aiming for the standard compiler command, I could run it with kscript and benefit from caching and its other features.

[1]: https://github.com/JetBrains/kotlin/blob/92d200e093c693b3c06e53a39e0b0973b84c7ec5/libraries/scripting/dependencies/src/kotlin/script/experimental/dependencies/annotations.kt#L31

**Comment by aartiPl** [comment]
Yes, your comment is entirely valid. My ultimate goal is to switch to Kotlin scripting backend while keeping all the excellent features of KScript (like interoperability with shell, caching, etc.). The only question is how much effort to put in to stay perfectly compatible, while in the long run, I will switch to Kotlin scripting anyway. Currently, in my opinion, it is not worthy of effort, as I would like to switch to the Kotlin scripting backend after version 4.2, which I am about to release. But still, if anyone wants to improve KScript with such features, I am happy to apply patches.
